tests/beaker_tests/lib/utilitylib.rb

```diff
@@ -923,8 +980,28 @@ def test_get(agent, filter, opt=:raw)
 def test_set(agent, cmd)
   return if cmd.empty?
   logger.info(cmd)
-  cmd_prefix = PUPPET_BINPATH + "resource cisco_command_config 'cc' "
-  on(agent, cmd_prefix + "test_set='#{cmd}'")
+  if !agent.nil?
+    cmd_prefix = PUPPET_BINPATH + "resource cisco_command_config 'cc' "
+    on(agent, cmd_prefix + "test_set='#{cmd}'")
+  else
+
+    conf_dir = `puppet config print | grep deviceconfig`
+    raise 'Puppet config not found' unless conf_dir
+
+    Puppet[:deviceconfig] = conf_dir.match(%r{(\S*)\n})[1]
```

Comment:
consider using HOCON to parse the config file. 
see: https://github.com/puppetlabs/puppet-resource_api/blob/master/lib/puppet/util/network_device/simple/device.rb

Reply:
Now using HOCON to parse config

Reply:
Removed, now using .cfg to define device details